src/utils/useHotspot.ts

```diff
@@ -184,6 +185,11 @@ const useHotspot = () => {
     }
 
     const response = await dispatch(fetchHotspotDetails(details))
+    if (
+      !(response.payload as AllHotspotDetails).onboardingRecord?.onboardingKey
```

Comment:
we now make sure we have an onboarding record and return false if not, triggering the navigation to the error screen